### PR TITLE
Add Better Checking For Execution API Methods

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Cleanup

**What does this PR do? Why is it needed?**

- [x] Adds a new helper and method to retrieve blocks by number.
- [x] Adds in better error checking. 

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/powchain/engine-api-client/v1/client.go [hunk 1]
```diff
@@ -199,9 +201,25 @@ func (c *Client) LatestExecutionBlock(ctx context.Context) (*pb.ExecutionBlock,
...
+	if err == nil && len(result.Number) == 0 {
+		err = ethereum.NotFound
+		return result, err
+	}
+	return result, handleRPCError(err)
+}
+
+// ExecutionBlockByHash fetches an execution engine block by hash by calling
+// eth_blockByHash via JSON-RPC.
+func (c *Client) ExecutionBlockByNumber(ctx context.Context, number *big.Int) (*pb.ExecutionBlock, error) {
```
Comment (terencechain) [inline beacon-chain/powchain/engine-api-client/v1/client.go:216]: Please add unit test

File: beacon-chain/powchain/engine-api-client/v1/client.go [hunk 2]
```diff
@@ -246,3 +268,14 @@ func handleRPCError(err error) error {
 		return err
 	}
 }
+
+func toBlockNumArg(number *big.Int) string {
```
Comment (terencechain) [inline beacon-chain/powchain/engine-api-client/v1/client.go:272]: Please add a test to this.  The pending case is worth testing

**Comment by rauljordan** [comment]
hi @nisdas is this one still relevant?

**Comment by nisdas** [comment]
Superseded by #11586

**Pull request closed without merging**